native/cocos/core/assets/Material.h

```diff
@@ -317,6 +317,7 @@ class Material : public Asset {
     bool uploadProperty(scene::Pass *pass, const ccstd::string &name, const MaterialPropertyVariant &val);
     void bindTexture(scene::Pass *pass, uint32_t handle, const MaterialProperty &val, uint32_t index = 0);
 
+    void prepareInfo(const cc::IMaterialInfo::DefinesType &patch, ccstd::vector<MacroRecord> &curr);
     template <typename T1, typename T2>
     void prepareInfo(const T1 &patch, ccstd::vector<T2> &cur) {
```

Comment:
There is already a `template` version of `prepareInfo` here . Why do we need an overload version with the same logic here?